### DISCUSSION TITLE
MaximumPacketSize property value must be in the range 1-268435455

### DISCUSSION THREAD

**Discussion by francois-le-ko4la** [emqx/emqx#14378] [General]
I push here my feedback on MaximumPacketSize parameter.

**Problem**

My EMQX:
- Environment: Docker
- Version: 5.8.3

After my update to EMQX & Home Assistant (HA) 2024.12.1, I lost my Zigbee devices. Here’s how I tried to resolve it:

-   Rebooting Z2M & EMQX → Didn’t work.
-   Docker Compose down/up (yeah, brute force...) → Still didn’t work.
-   Deleting MQTT and setting it up with basic config on HA → It worked (but… huh?).
-   Switching to advanced configuration and changing MQTT from 3.1.1 to 5 → It broke again (huh?!).

So, I put on my glasses and read the logs on HA:
```plaintext
2024-12-07 07:50:14.997 ERROR (Thread-40 (_thread_main)) [root] Uncaught thread exception
Traceback (most recent call last):
...
paho.mqtt.properties.MQTTException: MaximumPacketSize property value must be in the range 1-268435455
```

**Solution**

I reconfigured EMQX, setting the max packet size in bytes instead of my previous parameter (256MB).
→ Now it works with MQTT v5.

256 MB = 256 × 1024 × 1024 bytes = 268435456 bytes.
One byte over the limit. It worked fine before the update, but now it's stricter.

I'm sharing this in case it helps someone else!

**Comment by zmstone** [comment]
Thank you @francois-le-ko4la 
I consider it a EQMX side bug, will fix it to clamp the value in range`[1 ~ 268435456)` after config value parse.

**Reply by zmstone** [reply]
fixed in https://github.com/emqx/emqx/issues/14400 (5.8.5)

**Reply by francois-le-ko4la** [reply]
Thank you for your feedback and this fix !
